### ISSUE TITLE
Be able to see more than 1 aura per indicator

### ISSUE THREAD

**Issue by ChrisKayn** [brittyazel/EnhancedRaidFrames#33]
The main issue here is that if you want to use the anchor BOTTOM LEFT for debuffs for example, like really BIG, and want to track multiple debuffs or whatever, it will just show 1 one debuff, the one that you put first on the lines of that anchor. It will be incredibly good for configuration to add an option that multiple auras can be seen per anchor and they arrenge upwards, backwards, right, left, whatever.

**Comment by brittyazel** [comment]
I too want this, it's just going to take a while to implement it. I need to refactor a lot of the logic as it is right now to make this more feasible. Hang in there

**Comment by brittyazel** [comment]
The more I think about this, the more I wonder if it's necessary. Can you give me a use case where tracking multiple auras in a single spot is required over putting the auras into the 9 spots already provided? Do we need to track more than 9 things at once?

My concern is that the indicators are already very small as it is with 9, let alone if we start adding sub-indicators. The size would be so small to fit them all that it would effectively make the text and icon impossible to see

**Comment by ChrisKayn** [comment]
I remember I wanted this because theres no profile per spec or per class, so when I was using my paladin and then my priest I couldnt see some auras because they were using the same place, ,but yes u have a point, currently im happy with the addon as it is, but i think you should add something per character

**Comment by brittyazel** [comment]
There are profile's per spec and per class actually and have been for quite a while. You can find them under the "Profile" menu option

**Comment by ChrisKayn** [comment]
Oh great, do you think its possible to move the anchors though?

**Comment by brittyazel** [comment]
For tooltips?

**Comment by ChrisKayn** [comment]
No, like move the actual icon a little to the left or right for example on the party frame or raid frame

**Comment by brittyazel** [comment]
Yes that has been implemented for a few weeks now. Check your config panel

**Comment by ChrisKayn** [comment]
I am getting this error

757x RaidFrameIndicators\AuraIndicators.lua:166: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `GetName'
[string "@RaidFrameIndicators\AuraIndicators.lua"]:166: in function `UpdateIndicators'
[string "@RaidFrameIndicators\EnhancedRaidFrames.lua"]:58: in function <...ceRaidFrameIndicators\EnhancedRaidFrames.lua:58>
[string "=(tail call)"]: ?
[string "=[C]"]: in function `CompactUnitFrame_UpdateAuras'
[string "@FrameXML\CompactUnitFrame.lua"]:101: in function <FrameXML\CompactUnitFrame.lua:54>

Locals:
(*temporary) = ForbiddenNamePlate17UnitFrame {
 0 = <userdata>
 WidgetContainer = <unnamed> {
 }
 classificationIndicator = <unnamed> {
 }
 castBar = <unnamed> {
 }
 maxDispelDebuffs = 0
 inVehicle = false
 optionTable = <table> {
 }
 overAbsorbGlow = <unnamed> {
 }
 totalAbsorb = <unnamed> {
 }
 myHealPrediction = <unnamed> {
 }
 aggroHighlight = <unnamed> {
 }
 hideCastbar = true
 myHealAbsorbRightShadow = <unnamed> {
 }
 overHealAbsorbGlow = <unnamed> {
 }
 newUnit = true
 selectionHighlight = <unnamed> {
 }
 myHealAbsorbLeftShadow = <unnamed> {
 }
 ClassificationFrame = <unnamed> {
 }
 unit = "nameplate17"
 unitExists = true
 disableMouse = true
 otherHealPrediction = <unnamed> {
 }
 BuffFrame = <unnamed> {
 }
 name = <unnamed> {
 }
 totalAbsorbOverlay = <unnamed> {
 }
 myHealAbsorb = <unnamed> {
 }
 maxBuffs = 0
 maxDebuffs = 0
 RaidTargetFrame = <unnamed> {
 }
 healthBar = <unnamed> {
 }
 displayedUnit = "nameplate17"
 LoseAggroAnim = <unnamed> {

**Comment by brittyazel** [comment]
Please open a new ticket for issues not related this one

**Comment by Amissia** [comment]
By default, Blizzard will track 3 or so debuffs on frames to that bottom left anchor. In PvP this is useful because often times you pair it with a set of commands to actually extend the amount of debuffs shown per frame. Say I stop using Blizzard default tracking and use "Poison, Magic, Curse, etc." modifiers. Then a person has both Shadow Word: Pain and Vampiric Touch on them it would only show the one instead of pushing it to the side and displaying both magic debuffs. Same thing applies to something like full feral dots. I'm not sure if you support bleed tracking, but being able to see all the feral DoTs on a player at a single time is very valuable. In a PvE aspect, in the Workshop portion of Mechagon there are the mobs that put Arcing Zap on players. This debuff doesn't stack and is instead treated as an individual debuff with each application. Healers more than likely prioritize dispelling the person who has more than one instance of the debuff opposed to the person with just one. However, its hard to tell how many debuffs a person has if only one is available to view at a time. I think that's where the practical application of being able to see more than one aura lies.

**Comment by brittyazel** [comment]
Interesting. There are some limitations though, I.e. There's just not enough physical room on eachbindicstor to

**Comment by Amissia** [comment]
Sadly, I don't have the brain for something like that, but I am fairly good with ideas behind systems and not so much the systems themselves. 

I'm not sure if this is of any help to you,  but here are the scripts I run when I start my game and plan do PvP or do M+. You can probably make more sense of the scripts than I can, but I believe that they are essentially recreating the debuff anchor and unlocking the amount of debuffs shown. An issue that comes with just using this method for M+ is that with things like Griveous/Challengers burden it doesn't adjust debuffs accordingly and they end up being overwritten and falling under other anchors.

(They're run in this order as well)

Script 1:
/run function sp(f,i) tr="TOPRIGHT";f2=f.debuffFrames;s=f2[1]:GetWidth();f3=f2[i];f3:SetSize(s,s);f3:ClearAllPoints();if i>6 then f3:SetPoint("BOTTOMRIGHT",f2[i-3],tr,0,0) else f3:SetPoint(tr,f2[1],tr,-(s*(i-3)),0) end end

Script 2:
/run function CBF(f,i) bf=CreateFrame("Button",f:GetName().."Debuff"..i,f,"CompactDebuffTemplate");bf.baseSize=22;bf:SetSize(f.buffFrames[1]:GetSize()) end;function mv(f) for i=4,12 do sp(f,i) end end

Script 3:
/run function mv3(f) CompactUnitFrame_SetMaxDebuffs(f,12); if not f.debuffFrames[4] then for i=4,12 do CBF(f,i) end end mv(f) end;hooksecurefunc("CompactUnitFrame_UpdateDebuffs",function(f) if f:GetName():match("^Compact") then mv3(f) end end);

**Comment by steven89** [comment]
Hi, I'm also very interested in this feature, playing Rdruid, and in my case, i'd like to have `Rejuvenation, Germination, Regrowth` all in the same spot next to each other

What about letting user add extra `indicator`, where you select the anchor location (e.g. `Top Left`) and can set different horizontal and vertical offset.

A solution like this shouldn't require too much refactoring and geta rid of the "not enough room" problem, it basically let's user create new spots wherever they want.

**Comment by Amissia** [comment]
This definitely seems like it could be one approach to this feature, but as Britt said above:

> My concern is that the indicators are already very small as it is with 9, let alone if we start adding sub-indicators. The size would be so small to fit them all that it would effectively make the text and icon impossible to see

 For your case, assuming you aren't using all 9 anchors, or even if you are they do support horizontal and vertical offset with the recent updates. So you can easily put your HoTs all right next to one another. I know I'm literally no help when it comes to helping with the coding aspect, but I do think that keeping only 9 anchors and growing off of them should be the focus.

**Comment by emrus-main** [comment]
I made a fork of an old version (2.3.0) for just this purpose because I needed this for classic, since as a priest I had 10 different buffs to keep track of these to prevent overwriting other priests' buffs as the default UI would not show them.

It's a bit crude of an implementation using 6 preassigned indicators per corner, though it performs very well.

I will try to see if I can bring the changes up to version 3.1. However, the fort also has a number of other features mixed in; A combat toggle and a mouseover bug fix where the auras from this addon would block mouseover casts. So I will see if I can separate them into independent features, before a potential pull request.

**Comment by brittyazel** [comment]
Pull requests are welcome! I'm certain you'll have to modify your code a great deal, as it's effectively a different codebase since 2.3.

As to the main topic of this thread, I have ideas for having up to 4 indicators per position, leading to 36 possible indicators. Any more than 4 I feel would be extreme. With 4 we can split one indicator into a 2x2 grid, and the math to resize and reshape these mini-grids dynamically won't be too complex.

**Comment by MitsuhideA** [comment]
I don't know if you're still working on it, but it would be useful!

**Comment by martheli** [comment]
Any update on this feature? This is very helpful for mythic+. Id like to be able to track my buffs on party members around the frame which is great but I don't want to lose the default blizzard debuff implemtation of having 3 debuffs listed. Sometimes people get multiple debuffs on boss fights that are important to see at the same time.
